hooks/pre-exit

```diff
@@ -73,16 +73,22 @@ save-report-url() {
     echo "Could not get the tests report URL from $json_file. File not found."
     return
   fi
-  # Could be easier with jq, but we don't want to require it
+
   if which jq >/dev/null; then
-      echo "Using jq to parse the report URL"
-      report_url=$(jq -r '.run_url' "${json_file}")
+    echo "Using jq to parse the report URL"
+    report_url=$(jq -r '.run_url' "${json_file}" 2>&1) # Capture stderr for error reporting
+    if [[ "$report_url" == "null" || "$report_url" == "" ]]; then
+      echo "jq parsing failed with the message: $report_url"
+      echo "Contents of $json_file:"
+      cat "$json_file"
+      return
+    fi
```

Comment:
This block is similar to [91-93](https://github.com/buildkite-plugins/test-collector-buildkite-plugin/pull/74/files#diff-28060270dece4fee8ea71b3090f651edfe4da2325cca76c084c1ac8f6ffc5b22R91-R93) So they should be unified. 
This change on the other block should be enough.

Reply:
done